### ISSUE TITLE
Wacom touchscreen etc. absolute coordinate mapping displaces pointer

### ISSUE THREAD

**Issue by piater** [any1/wayvnc#85]
My laptop has a Wacom touchscreen. Sway configures libinput to map touch coordinates (stylus and finger) to the corresponding screen area, which by default includes the entire virtual screen area covered by all displays. To keep the pointer under the stylus/finger in the presence of external displays, I `map_to_output` my Wacom devices to the internal display.

When I use wayvnc to map an external display to a VNC viewer window, this coordinate mapping falls apart. Here is what I think is going on: VNC transports the absolute pointer coordinates to the attached display and places the pointer in the correct location. However, this contradicts the `map_to_output` setting, which somehow kicks in and moves the pointer back to the internal display, at coordinates that appear to correspond to those of the external display. The result is that the pointer flickers back and forth between the external display (at the correct location) and the internal display, possibly outside the VNC viewer window.

I confirmed the mapping issue by configuring `map_from_region` to roughly the internal screen area covered by the VNC image and `map_to_output` to the external display captured by wayvnc. In this case no pointer flicker occurs, and the stylus behaves as it should.

The same mapping issue arises with finger touch events but the manifestations are different because they do not move the mouse pointer.

I am not sure how to address this issue, or even whether wayvnc is the right place to fix it. But the observation that coordinates within the VNC-attached display are mapped back via `map_to_output` makes me think that wayvnc may have the power to break this cycle.

**Comment by any1** [comment]
This looks like a sway issue. Have you tried running wayvnc on a different seat?

A new seat can be created in sway by issuing the command `swaymsg seat wayvnc fallback false` (see man sway-input).
Then start wayvnc with the seat argument: `wayvnc --seat=wayvnc`.

Not all wayland clients support multi-seat, so YMMV, but this might give us some insight into the problem.

**Comment by piater** [comment]
TL;DR: Running wayvnc on a different seat solves it, many thanks! You can close this bug.

Now I'd like to understand what's going on. How accurate is the following:

If I run wayvnc on the fallback (or only) seat and I move the pointer into the VNC client image, the pointer is warped over to the VNC-backed display where it continues to be controlled by seat0 devices. Touch creates the above inconsistency because touch event coordinates are mapped back to the internal display.

If I run wayvnc on a separate seat it has its own pointer, so the local pointer is not warped over. The wayvnc pointer follows the local pointer, but it has no power over the local pointer.
